src/helpers/getSubdivision.ts

```diff
@@ -0,0 +1,20 @@
+import { TGeographySubdivision } from "@/types";
+
+const findSubdivision = (search: string, dataSet: TGeographySubdivision[]) => {
+  let subdivision = dataSet.find((c) => c.code.toLowerCase() === search.toLowerCase());
+  if (!subdivision) subdivision = dataSet.find((c) => c.name.toLowerCase() === search.toLowerCase());
```

Comment:
Request: Is there a way this can avoid potentially iterating over the full data set twice?